core/domain/stats_services.py

```diff
@@ -178,3 +223,155 @@ def get_exploration_stats(exploration_id, exploration_version):
             } for state_name in exploration.states
         },
     }
+
+
+def record_answer(exploration_id, exploration_version, state_name,
+                  handler_name, rule_str, session_id, time_spent_in_sec,
+                  params, answer_value):
+    """
+    Record an answer by storing it to the corresponding StateAnswers entity.
+    """
+    # Retrieve state_answers from storage
+    state_answers = get_state_answers(
+        exploration_id, exploration_version, state_name)
+
+    # Get interaction id from state_answers if it is stored there or
+    # obtain it from the exploration (note that if the interaction id 
+    # of a state is changed by editing the exploration then this will
+    # correspond to a new version of the exploration, for which a new
+    # state_answers entity with correct updated interaction_id will
+    # be created when the first answer is recorded).
+    # If no interaction id exists, use None as placeholder.
+    interaction_id = None
+    if state_answers is not None:
+        interaction_id = state_answers.interaction_id
+
+    if interaction_id is None:
+        # retrieve exploration and read its interaction id
+        exploration = exp_services.get_exploration_by_id(
+            exploration_id, version=exploration_version)
+        if exploration.states[state_name].interaction.id:
```

Comment:
I think you are right, it should always have an interaction id. But not sure really. Maybe the idea was to be able to migrate old answers in case they didn't have any interaction id? I don't know if this is relevant though.